src/App/Fossa/Config/ConfigFile.hs

```diff
@@ -195,6 +197,8 @@ data ConfigFile = ConfigFile
   , configExperimental :: Maybe ExperimentalConfigs
   , configVendoredDependencies :: Maybe VendoredDependencyConfigs
   , configTelemetry :: Maybe ConfigTelemetry
+  , configCustomLicenseSearch :: Maybe (NonEmpty ConfigGrepEntry)
+  , configKeywordSearch :: Maybe (NonEmpty ConfigGrepEntry)
```

Comment:
Since this was marked resolved but unchanged - is the intention "if it is provided, it should be non-empty, but it's okay if it's not provided at all"?

If so:
- Why does it matter? Is the intention to catch a syntax mistake in the config?
  - I don't really see why we care whether the user wants to put an empty array in the config, but eh, if that's something we want to stop then I won't block on it.
- Please add a comment describing this to make it more clear why we're doing this.

Reply:
Sorry, I think I just clicked too fast when resolving a bunch of issues and then missed fixing this.

These still need to remain `Maybe [ConfigGrepEntry]` so that there's no error when the config is absent, but I've gotten rid of the `NonEmpty`.

Reply:
> These still need to remain Maybe [ConfigGrepEntry] so that there's no error when the config is absent

I recommend instead deserializing with I think `.:?`- a `Maybe []` seems equivalent to `[]` in this instance to me.

Ultimately though not a huge deal if you want to keep it as is.